lib/turn.rb

```diff
@@ -0,0 +1,78 @@
+class Turn
+  attr_reader :player1, :player2, :spoils_of_war, :type, :winner
+
+  def initialize(player1, player2)
+    @player1 = player1
+    @player2 = player2
+    @spoils_of_war = []
+    @winner = winner
+  end
+
+  def type
+      if player1.deck.rank_of_card_at(0) != player2.deck.rank_of_card_at(0)
+        :basic
+      elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) != player2.deck.rank_of_card_at(2)
+        :war
+      elsif player1.deck.rank_of_card_at(0) == player2.deck.rank_of_card_at(0) && player1.deck.rank_of_card_at(2) == player2.deck.rank_of_card_at(2)
+        :mutually_assured_destruction
+      end
+  end
+
+  def winner
+    if type == :basic
+      if player1.deck.rank_of_card_at(0) > player2.deck.rank_of_card_at(0)
+        @player1
+      else
+        @player2
+      end
+      #return the player with the highest ranking card at index position 0
+    elsif type == :war
+      if player1.deck.rank_of_card_at(2) > player2.deck.rank_of_card_at(2)
+        @player1
+      else
+        @player2
+      end
+    elsif type == :mutually_assured_destruction
+      "No Winner"
+    end
+  end
+
+  def pile_cards
+      if type == :basic
+      #each player will send one card (the top card) to the spoils pile
+      # spoils_of_war = []
+      spoils_of_war << player1.deck.remove_card
+      spoils_of_war << player2.deck.remove_card
+      # spoils_of_war
+      end
+
+      if type == :war
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player1.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+        spoils_of_war << player2.deck.remove_card
+      end
+
+      if type == :mutually_assured_destruction
+        player1.deck.remove_card
+        player1.deck.remove_card
+        player1.deck.remove_card
+        player2.deck.remove_card
+        player2.deck.remove_card
+        player2.deck.remove_card
+      end
+    # elsif type == :war
+    #   #each player will send 3 cards (the top 3 cards) to the spoils pile
+    #
+    # elsif type == :mutually_assured_destruction
+    #   #each player will remove three cards from play (the top three cards in their deck).
+    #   #These cards are NOT send to the spoils pile, they are simply removed from each players deck
+  end
+
+  def award_spoils
+    @winner.deck.cards.concat(spoils_of_war)
+    #research concat
+    end
```

Comment:
This method is wonderfully written and works as intended if the game's are of a `:basic` or `:war` type. If the game is `:mutually_assured_destruction` the game breaks here. This is because, if you remember, when a game is `:mutually_assured_destruction` we get `"No Winner"` returned from the `winner` method. And therefore when we call `award_spoils` and it goes to get `@winner`, it will have the value `"No Winner"` string and you cannot call `.deck` on a String.

One way to avoid this behavior is to add a conditional statement to this method so that way it only runs if an actual player object is stored in `@winner`.

```ruby
def award_spoils
  if @winner != "No Winner" #<== If winner is not equal to "No Winner"
    @winner.deck.cards.concat(spoils_of_war) #<== then do this
  end #<== otherwise do nothing
end
```

===

One other thing. According to the interaction pattern this method is supposed to take in an argument but your method doesn't. I imagine the method would still work as written using the `@winner` instance variable, but we want to make sure that when we are given a specification like the interaction, we try and stick to it exactly. 🙂 

```ruby
def award_spoils(winner)
  # code...
end
```